### PR TITLE
Rick/merge v1.1.0

### PR DESCRIPTION
**Pull request by rickpasetto**
Includes changes for LOOP-3830 and LOOP-3856.

~Note: some reworking of `DeliveryLimitsEditor` had to be done because now it is doing both a `maxTempBasalSavePreflight` _and_ a `syncDeliveryLimits`.~

Refactor everything into new `TherapySettingsViewModelDelegate`:
The number of function aliases in `TherapySettingsViewModel` was getting to be a mess.  I consolidated `SaveCompletion`, `MaxTempBasalSavePreflight`, `SyncBasalRateSchedule`, and `SyncDeliveryLimits` all into a new `TherapySettingsViewModelDelegate`. Things got a bit cleaner, and now `DeviceDataManager` handles doing both the `maxTempBasalSavePreflight` and `syncDeliveryLimits` so it is not so loose of a contract.

 @ps2 would like you to review...

### CODE REVIEW

File: LoopKitUI/ViewModels/TherapySettingsViewModel.swift [hunk 1]
```diff
@@ -11,37 +11,37 @@ import LoopKit
 import HealthKit
 import SwiftUI
 
-public typealias SyncSchedule = (_ items: [RepeatingScheduleValue<Double>], _ completion: @escaping (Swift.Result<BasalRateSchedule, Error>) -> Void) -> Void
-public typealias SyncDeliveryLimits = (_ deliveryLimits: DeliveryLimits, _ completion: @escaping (_ result: Swift.Result<DeliveryLimits, Error>) -> Void) -> Void
+public enum SaveTherapySettingsError: Error {
+    case cancelTempBasalError(Error)
+    case missingPumpManager
```
Comment (ps2) [inline LoopKitUI/ViewModels/TherapySettingsViewModel.swift]: Maybe this enum goes away, if we don't need the missingPumpManager case (see other PR).

File: LoopKitUI/ViewModels/TherapySettingsViewModel.swift [hunk 2]
```diff
@@ -11,37 +11,37 @@ import LoopKit
 import HealthKit
 import SwiftUI
 
-public typealias SyncSchedule = (_ items: [RepeatingScheduleValue<Double>], _ completion: @escaping (Swift.Result<BasalRateSchedule, Error>) -> Void) -> Void
```
Comment (ps2) [inline LoopKitUI/ViewModels/TherapySettingsViewModel.swift:14]: Thanks for moving these to an even more appropriate spot, now that the delegate is defined.

File: LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift [hunk 3]
```diff
@@ -379,13 +392,39 @@ public struct DeliveryLimitsEditor: View {
...
+            default:
+                return SwiftUI.Alert(
+                    title: Text(LocalizedString("Unable to Save", comment: "Alert title when error occurs while saving a schedule")),
+                    message: Text(error.localizedDescription)
+                )
+            }
         }
     }
 
+    private func cancelTempBasalAlert(_ error: Error?) -> SwiftUI.Alert {
```
Comment (ps2) [inline LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift]: There are many reasons why saving the max temp might fail, cancelling basal is just one of them. For the other errors, we'll need pass an `Error` with a sufficient `localizedDescription` (and possibly recovery suggestion). What is the reason this has a special type? Would recommend having a unified handling of errors here, unless there is a specific and clear UX need for this one to be different.

File: LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift [hunk 4]
```diff
@@ -379,13 +392,39 @@ public struct DeliveryLimitsEditor: View {
...
+        Alert(
+            title: Text(LocalizedString("Failed to Cancel Temp Basal", comment: "Alert title for failing to cancel temp basal")),
+            message: Text(String(format: LocalizedString("%@Tidepool Loop was unable to cancel your current temporary basal rate, which is higher than the new Max Basal limit you have set. This may result in higher insulin delivery than desired.\n\nConsider suspending insulin delivery manually and then immediately resuming to enact basal delivery with the new limit in place.",
```
Comment (ps2) [inline LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift]: App specific string here. Would recommend, as per the above comment, that this special handling for the error be removed and instead, have the hosting app return a sufficiently descriptive error type.

**Review by ps2 — CHANGES_REQUESTED** [review]
The noPumpManager error should be cleaned up, and there is an app-specific string.

File: LoopKitUI/ViewModels/TherapySettingsViewModel.swift (hunk 1 above)
Comment (rickpasetto) [reply LoopKitUI/ViewModels/TherapySettingsViewModel.swift]: Sure, but we still need *some* error meaning "CancelTempBasal failed", wrapping around the reason for it to fail.

File: LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift (hunk 3 above)
Comment (rickpasetto) [reply LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift]: There is a specific UX the team wants separate from the save errors (and which already was implemented in v1.1.0).  See [LOOP-3856](https://tidepool.atlassian.net/browse/LOOP-3856) for more details.

File: LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift (hunk 4 above)
Comment (rickpasetto) [reply LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift]: As above, see the bug.  There's a different alert title and content.

File: LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift (hunk 3 above)
Comment (rickpasetto) [reply LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift]: Moved content to `DeviceDataManager` in `Loop`

File: LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift (hunk 4 above)
Comment (rickpasetto) [reply LoopKitUI/Views/Settings Editors/DeliveryLimitsEditor.swift]: Moved content to `DeviceDataManager` in `Loop`

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**